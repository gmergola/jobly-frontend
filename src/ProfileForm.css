.Profile-container {
  background-color: rgb(213, 224, 235);
  border-radius: 50px;
  color: rgb(75, 86, 139);
}

.Profile-photo {
  border-radius: 180%;
}

.ProfileForm-button {
  margin-top: 20px;
  width: 100px;
  margin-bottom: 20px;
}

.Profile-input {
  width: 500px;
  border: none;
}

.input-group-text {
  margin-left: 2rem;
}

@media only screen and (max-width: 600px) {
  .Profile-input {
    width: 65%;
  }

  .input-group-text {
    margin-left: 0rem;
  }
}