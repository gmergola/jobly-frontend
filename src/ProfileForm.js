import React, { useState } from 'react';
import defaultPhoto from './images/default-user-photo.png';
import Alert from './Alert';
import JoblyApi from "./HelperApi";
import './ProfileForm.css';

/**ProfileForm: Edits a user's information as long as the password is corrent */
function ProfileForm({ currentUser, setCurrentUser }) {
  const [errorMessage, setErrorMessage] = useState([]);
  const [formData, setFormData] = useState({
    first_name: currentUser.first_name || "",
    last_name: currentUser.last_name || "",
    email: currentUser.email || "",
    photo_url: currentUser.photo_url || "",
    username: currentUser.username,
    password: "",
  });

  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData(f => ({
      ...f,
      [name]: value,
      errors: []
    }));
  }

  async function handleSubmit(evt) {
    evt.preventDefault();
    let profileData = {
      first_name: formData.first_name || undefined,
      last_name: formData.last_name || undefined,
      email: formData.email || undefined,
      photo_url: formData.photo_url || undefined,
      password: formData.password
    };

    let username = formData.username;
    let editedUser;

    try {
      editedUser = await JoblyApi.updateProfile(username, profileData);
    } catch (errors) {
      setErrorMessage(errors);
      return;
    }

    setFormData(f => ({ ...f, password: "" }));
    setErrorMessage([]);
    setCurrentUser(editedUser);
  }

  return (
    <div>
      <div>
        <h1>{currentUser.username}</h1>
        <p>User's name: {currentUser.first_name} {currentUser.last_name}</p>
        {currentUser.photo_url ? <img src={currentUser.photo_url} alt="default" /> :
          <img src={defaultPhoto} alt="default" />}
      </div>
      <br />
      <div>
      <br />
      <h3>Edit your profile information</h3>
      <div ><Alert errors={errorMessage} /></div>
        <form className="form-group" onSubmit={handleSubmit}>
          <div className="input-group">
            <div className="input-group-prepend">
              <span className="input-group-text">First Name </span>
            </div>
            <input className="form-control" onChange={handleChange} value={formData.first_name} name="first_name"></input>
          </div>
          <br />
          <div className="input-group">
            <div className="input-group-prepend">
              <span className="input-group-text">Last Name </span>
            </div>
            <input className="form-control" onChange={handleChange} value={formData.last_name} name="last_name"></input>
          </div>
          <br />
          <div className="input-group">
            <div className="input-group-prepend">
              <span className="input-group-text">Email </span>
            </div>
            <input className="form-control" onChange={handleChange} value={formData.email} name="email"></input>
          </div>
          <br />
          <div className="input-group">
            <div className="input-group-prepend">
              <span className="input-group-text">Re-Enter Password </span>
            </div>
            <input className="form-control" onChange={handleChange} value={formData.password} name="password"></input>
          </div>
          <br />
          <div className="input-group">
            <div className="input-group-prepend">
              <span className="input-group-text">Photo URL </span>
            </div>
            <input className="form-control" onChange={handleChange} value={formData.photo_url} name="photo_url"></input>
          </div>
          <br />
          <button className="ProfileForm-button btn btn-info" type="submit">Save</button>
        </form>
      </div>
    </div>
  )

}

export default ProfileForm;