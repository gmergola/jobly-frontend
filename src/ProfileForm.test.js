import React from "react";
import { render } from "@testing-library/react";
import ProfileForm from "./ProfileForm";
import { MemoryRouter } from "react-router";


it("renders without crashing", function() {
  render(
    <MemoryRouter>
        <ProfileForm />
    </MemoryRouter>
  );
});

it("matches snapshot", function() {
  const { asFragment } = render(
    <MemoryRouter>
      <ProfileForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});