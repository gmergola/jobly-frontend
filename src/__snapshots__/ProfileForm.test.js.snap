// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1 />
      <p>
        User's name:  
      </p>
      <img
        alt="default"
        src="default-user-photo.png"
      />
    </div>
    <br />
    <div>
      <br />
      <h3>
        Edit your profile information
      </h3>
      <div />
      <form
        class="form-group"
      >
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              First Name 
            </span>
          </div>
          <input
            class="form-control"
            name="first_name"
            value=""
          />
        </div>
        <br />
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              Last Name 
            </span>
          </div>
          <input
            class="form-control"
            name="last_name"
            value=""
          />
        </div>
        <br />
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              Email 
            </span>
          </div>
          <input
            class="form-control"
            name="email"
            value=""
          />
        </div>
        <br />
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              Re-Enter Password 
            </span>
          </div>
          <input
            class="form-control"
            name="password"
            value=""
          />
        </div>
        <br />
        <div
          class="input-group"
        >
          <div
            class="input-group-prepend"
          >
            <span
              class="input-group-text"
            >
              Photo URL 
            </span>
          </div>
          <input
            class="form-control"
            name="photo_url"
            value=""
          />
        </div>
        <br />
        <button
          class="ProfileForm-button btn btn-info"
          type="submit"
        >
          Save
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
